### PR TITLE
Typescript typings & bump dependencies

### PR DESCRIPTION
**Pull request by aldy505**
Hi!

I've come across this repo while I was browsing your entire repo, originally to find something as an alternative to sanitize-html. But I stumbled upon this one, got interested in how simple this is compared to Tempura (and yes I also use Tempura).

I might use this one on my current project as an alternative to tempura because I don't need stuff that's as powerful as tempura, and the project's in Typescript, so that's why I added these typings.

Thanks!

### CODE REVIEW

**Review by lukeed — CHANGES_REQUESTED** [review]
Hi, thanks! Please remove the dependency upgrades & the `package-lock.json` file. You may adjust the `.gitignore` to include the `*-lock.*` pattern. This PR should only include TS definitions

File: builder.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 const fs = require('fs');
-const mkdir = require('mk-dirs');
-const imports = require('rewrite-imports');
+const { mkdir } = require('mk-dirs');
+const { rewrite } = require('rewrite-imports');
```
Comment (lukeed) [inline builder.js]:
```suggestion
const mkdir = require('mk-dirs');
const imports = require('rewrite-imports');
```

File: builder.js [hunk 2]
```diff
@@ -9,5 +9,5 @@ mkdir('dist').then(_ => {
 	// Copy as is for ESM
 	fs.writeFileSync(pkg.module, data);
 	// Rewrite ESM ~> CJS
-	fs.writeFileSync(pkg.main, imports(data).replace('export default', 'module.exports ='));
+	fs.writeFileSync(pkg.main, rewrite(data).replace('export default', 'module.exports ='));
```
Comment (lukeed) [inline builder.js]:
```suggestion
	fs.writeFileSync(pkg.main, imports(data).replace('export default', 'module.exports ='));
```

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export default function<T extends { [key: string]: any }>(str: string, mix: T): string;
```
Comment (lukeed) [inline index.d.ts]:
```suggestion
export type Values = Record<string, any> | any[];
export default function<T extends Values>(template: string, values: T): string;
```

[Playground](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBDAnmYcBqBDANgV2AZzgF44AlYAY2gBMAefGKASwDsBzAGjgxcQD44AH268A2gF0A3AChqlLBiioAZjhYUYTCCwTAAtmCxMYwWgBU4oEy2qFMuAnwAUJgwpMAuOA2bsuAN2w8fC8zAEovH1Y2GWlXQ2NgJwByAAlgLCwILgBvHJYMPWAAX2KAQmTcuAKir2SAd2gsamS4YrCZAHpOogF0zOy4Rqhmsuk4-QSTFNSIeupEXJyABlKyuDyARlLKuFFksEUYFmAoXeTAPg3AAl3k8Q7pbt64WfnFuEPYE6h16-H4o2myWUEAgXgARHlgRBSmDJHAAEaKcF5RFQGGSXY5OBQrybABMAGY2vdHgIcXAwfiCbCEUiKbC-pMAUkAAbSOBwAByhWAXjyNWAADoFAxSksBYLlEwoKLiuy4AAZCAUDCabR8nIYajUJT4fCCijGRClOBOPJanUEfVUNSMY3teWzeHwpgEDUACwgztd+tWxSWnu9BEF239GxygZdwbxpWkLNy8oFfPlHKlMpgdQVOAA1sBKim4CKM3BkgBRaj1RS2ZLy-3yi26kIbAuGpCZiCEACC7AyBHzHI5NpYdrqAFUAMqdmscuscyM+rz7YCq874LDAYBgc5KFAr8TSdpdHoCbm1ODlytQWxcLO5h7HxXK1VaFheJVdnvrwhOCedsL3p4nSjJtlxgLg1w3MAuB3UCgA)

**Review by lukeed — CHANGES_REQUESTED** [review]

File: builder.js (hunk 1 above)
Comment (aldy505) [reply builder.js]: Oh yeah, sorry, forgot about this.

**Review by lukeed — APPROVED** [review]
Thanks!

**Pull request merged**